### PR TITLE
Caps logging about not being able to push to amount of desired slaves

### PR DESCRIPTION
**Pull request by tinwelint**
to max once per 5 seconds, as to not fill up the log

### CODE REVIEW

**Comment by digitalstain** [comment]
@tinwelint The failure seems relevant to the change

**Comment by tinwelint** [comment]
@digitalstain fixed now

File: enterprise/ha/src/main/java/org/neo4j/kernel/ha/transaction/TransactionPropagator.java [hunk 1]
```diff
@@ -72,6 +74,20 @@
         }
     }
 
+    private static class PushedToTooFewSlavesContext
```
Comment (digitalstain) [inline enterprise/ha/src/main/java/org/neo4j/kernel/ha/transaction/TransactionPropagator.java]: @tinwelint This class is not used anywhere

**Comment by tinwelint** [comment]
@digitalstain removed it and also added some javadoc to `TransactionPropagator`.

**Pull request merged**